nextcore/endpoint/interactions.py

```diff
@@ -0,0 +1,158 @@
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+from __future__ import annotations
+
+import asyncio
+
+from aiohttp import web
+from json import JSONDecodeError
+from logging import getLogger
+from nacl.exceptions import BadSignatureError
+from nacl.signing import VerifyKey
+from typing import TYPE_CHECKING
+
+from nextcore.common import Dispatcher
+
+
+if TYPE_CHECKING:
+    from typing import Final
+
+    from discord_typings import InteractionData
+    from multidict import CIMultiDictProxy
+
+
+logger = getLogger(__name__)
+
+
+__all__: Final[tuple[str, ...]] = (
+    "InteractionEndpoint",
+    "verify_signature",
+)
+
+
+def verify_signature(public_key: str, signature: str, timestamp: str, body: str) -> bool:
+    verify_key = VerifyKey(bytes.fromhex(public_key))
+    verify_key.verify(f"{timestamp}{body}".encode(), bytes.fromhex(signature))
+    return True
+
+
+class InteractionEndpoint:
+    _public_key: str | None
+    _route: str | None
+    event_dispatcher: Dispatcher
+
+    # def __init__(
+    #         self,
+    #         *,
+    #         web_app: web.Application = None,
+    #         route_path: str = "/interactions/test"
+    # ):
+    #     self.public_key: str | None = None
+    #     self.app: web.Application = web_app or web.Application(middlewares=[interaction_endpoint("asdf", "/interactions/test")])
+    #     self.site: web.TCPSite | None = None
+    #     self.event_dispatcher: Dispatcher = Dispatcher()
+    #
+    #     self.app.add_routes([
+    #         web.route("POST", route_path, self._on_interaction_incoming)
+    #     ])
+
+    def __init__(self):
+        self._public_key = None
+        self._route = None
+        self.event_dispatcher = Dispatcher()
+
+    def _verify_from_headers(self, headers: CIMultiDictProxy, body: str) -> bool:
+        try:
+            signature: str = headers.get("x-signature-ed25519")
+            timestamp: str = headers.get("x-signature-timestamp")
```

Comment:
If these are guaranteed to return `str`, then why not use brackets syntax instead of `dict.get`?